### PR TITLE
Complete http client

### PR DESCRIPTION
**Pull request by milch**
Note: waiting for dropbox/djinni#269 to be fixed

### CODE REVIEW

File: src/jsonify_middleware.cpp [hunk 1]
```diff
@@ -20,6 +25,14 @@ MiddlewareResponse JsonifyMiddleware::handle_response(Response &response) {
 }
 
 MiddlewareRequest JsonifyMiddleware::handle_request(Request &request) {
+    let jsonBody = request.get_component<JsonComponent>("bodyJson");
+    
+    if (jsonBody) {
+        let jsonString = (*jsonBody).dump();
```
Comment (milch) [inline src/jsonify_middleware.cpp]: use `->`

File: src/http_client.cpp [hunk 2]
```diff
@@ -4,23 +4,84 @@
...
+void HTTPClient::patch(const string &url, success_block_t success, error_block_t error) {
+    this->send(HTTPMethod::PATCH, url, nullopt, nullopt, success, error);
+}
+
+void HTTPClient::patch(const string &url, json11::Json &json, success_block_t success, error_block_t error) {
+    this->send(HTTPMethod::GET, url, nullopt, json, success, error);
+}
+
+void HTTPClient::patch(const string &url, unordered_map<string, string> parameters, success_block_t success, error_block_t error) {
+    this->send(HTTPMethod::GET, url, parameters, nullopt, success, error);
```
Comment (milch) [inline src/http_client.cpp]: Why is the HTTPMethod GET almost everywhere?

Comment (heinzl) [inline src/http_client.cpp]: not so smart copy/paste

**Comment by milch** [comment]
👍🏼

**Pull request merged**